File: laba4/C/main.c
#include <stdio.h>
#include <stdbool.h>

int main(void) {
    int c;
    int char_count = 0; 
    int line_count = 0; 
    int count_word = 0;

    bool is_word = false;

    while ((c = getchar()) != EOF) {
        char_count++; 

     
        if (c == '\n') {
            line_count++; 
        }

        if (c == ' ' || c == '\n' || c == '\t') {
            if (is_word) {
                count_word++; 
                is_word = false; 
            }
        } 
        else {
            is_word = true; 
        }
    }

 
    if (is_word) {
        count_word++;
    }
    printf("\n");
    printf("Lines: %d\n", line_count);      
    printf("Characters: %d\n", char_count); 
    printf("Words: %d\n", count_word);  

    return 0;
}

File: laba4/rust/main.rs
use std::io::{self, BufRead};

fn main() {
    
    let stdin = io::stdin(); 
    let handle = stdin.lock(); 

    let mut char_count = 0; 
    let mut line_count = 0; 
    let mut word_count = 0; 
    let mut is_word = false; 

    for line in handle.lines() {
        let line = line.unwrap(); 
        line_count += 1; 

        for c in line.chars() {
            char_count += 1; 
            if c.is_whitespace() { 
                if is_word {
                    word_count += 1; 
                    is_word = false; 
                }
            } else {
                is_word = true; 
            }
        }

        
        if is_word {
            word_count += 1;
            is_word = false; 
        }
        
       
    }
    println!("");
    println!("Lines: {}", line_count);
    println!("Characters: {}", char_count);
    println!("Words: {}", word_count);
}